contrib/mesos/pkg/scheduler/scheduler.go

```diff
@@ -923,6 +924,10 @@ func (ks *KubernetesScheduler) recoverTasks() error {
 		ks.slaves.checkAndAdd(slaveId, t.Offer.Host())
 	}
 	for _, pod := range podList.Items {
+		if _, isMirrorPod := pod.Annotations[kubelet.ConfigMirrorAnnotationKey]; isMirrorPod {
+			// mirrored pods are never reconciled
```

Comment:
before this patch, recoverTasks lists **all** pods across **all**
namespaces and attempts to reconcile them against it's registry. mirrored
pods (static pods) never pass through the scheduler.

On Mon, Sep 14, 2015 at 9:18 AM, Dr. Stefan Schimanski <
notifications@github.com> wrote:

> In contrib/mesos/pkg/scheduler/scheduler.go
> <https://github.com/kubernetes/kubernetes/pull/13918#discussion_r39391628>
> :
>
> > @@ -923,6 +924,10 @@ func (ks *KubernetesScheduler) recoverTasks() error {
> >  		ks.slaves.checkAndAdd(slaveId, t.Offer.Host())
> >  	}
> >  	for _, pod := range podList.Items {
> > +		if _, isMirrorPod := pod.Annotations[kubelet.ConfigMirrorAnnotationKey]; isMirrorPod {
> > +			// mirrored pods are never reconciled
>
> why? // mirrored pods are not managed by the scheduler
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/kubernetes/kubernetes/pull/13918/files#r39391628>.
>